### PR TITLE
refactor(kzip info): allow composing other operations in kzip.Scan

### PR DESCRIPTION
**Pull request by justbuchanan**
Scanning a kzip can be slow for large files, so if multiple operations
need to be done, it's preferable that they all be done in one
kzip.Scan() instead of separately. This change moves most kzip info
logic into an "Accumulator" class that can be called from your own
invocation of kzip.Scan().

The motivation for this is that I need to get summary info for a kzip
and also search for certain units by file path. Doing these in one
kzip.Scan() saves a lot of time.

### CODE REVIEW

File: kythe/go/platform/kzip/info/info.go [hunk 1]
```diff
@@ -29,58 +29,85 @@ import (
...
-				sourceInfo(riCorpus, cuLang, kzipInfo).Count++
-				srcCorpora.Add(riCorpus)
-				srcsWithRI.Add(ri.Info.Path)
-			}
-		}
-		srcsWithoutRI := srcs.Diff(srcsWithRI)
-		for path := range srcsWithoutRI {
-			msg := fmt.Sprintf("source %q in CU %v doesn't have a required_input entry", path, u.Proto.GetVName())
-			kzipInfo.CriticalKzipErrors = append(kzipInfo.CriticalKzipErrors, msg)
+// NewAccumulator creates a new Accumulator instance.
```
Comment (salguarnieri) [inline kythe/go/platform/kzip/info/info.go]: Can you add a doc for what units filesize should be?

File: kythe/go/platform/kzip/info/info.go [hunk 2]
```diff
@@ -29,58 +29,85 @@ import (
...
+func NewAccumulator(fileSize int64) *Accumulator {
```
Comment (salguarnieri) [inline kythe/go/platform/kzip/info/info.go]: Do Accumulator, NewAccumlator, etc. need to be exported yet? Or can that wait until your future changes that need them?

Comment (justbuchanan) [reply kythe/go/platform/kzip/info/info.go]: The changes that need them are in google3.

File: kythe/go/platform/kzip/info/info.go (hunk 1 above)
Comment (justbuchanan) [reply kythe/go/platform/kzip/info/info.go]: Done.

File: kythe/go/platform/kzip/info/info.go [hunk 3]
```diff
@@ -29,58 +29,86 @@ import (
...
-				sourceInfo(riCorpus, cuLang, kzipInfo).Count++
-				srcCorpora.Add(riCorpus)
-				srcsWithRI.Add(ri.Info.Path)
-			}
-		}
-		srcsWithoutRI := srcs.Diff(srcsWithRI)
-		for path := range srcsWithoutRI {
-			msg := fmt.Sprintf("source %q in CU %v doesn't have a required_input entry", path, u.Proto.GetVName())
-			kzipInfo.CriticalKzipErrors = append(kzipInfo.CriticalKzipErrors, msg)
+// NewAccumulator creates a new Accumulator instance given the kzip file size
```
Comment (salguarnieri) [inline kythe/go/platform/kzip/info/info.go]: nit, if you ruse the param name (fileSize) it will register in godoc.

Comment (justbuchanan) [reply kythe/go/platform/kzip/info/info.go]: :+1:

**Pull request merged**